lib/burnside/convert-expr.cc

```diff
@@ -551,7 +558,32 @@ class ExprLowering {
     return gen(funRef);
   }
   template<typename A> M::Value *genval(const Ev::FunctionRef<A> &funRef) {
-    TODO();
+    TODO();  // Derived type functions (user + intrinsics)
+  }
+  template<Co::TypeCategory TC, int KIND>
+  M::Value *genval(const Ev::FunctionRef<Ev::Type<TC, KIND>> &funRef) {
+    if (const auto &intrinsic{funRef.proc().GetSpecificIntrinsic()}) {
+      std::string name{intrinsic->name};
+      M::Type ty{genTypeFromCategoryAndKind(builder.getContext(), TC, KIND)};
+      // Probe the intrinsic library
+      if (std::optional<M::FuncOp> func{
+              intrinsics.getFunction(name, ty, builder)}) {
+        L::SmallVector<M::Value *, 2> operands;
+        for (const auto &arg : funRef.arguments()) {
+          if (auto *expr{Ev::UnwrapExpr<Ev::Expr<Ev::SomeType>>(arg)}) {
+            operands.push_back(genval(*expr));
+          } else {
+            TODO();  // Weird arguments, can it happen at that stage ?
```

Comment:
Are there any intrinsics that can take optional arguments or? If not, the `if` could go away and just use an `assert` here.

Reply:
There are, though I do not think the numerical ones do. The ones that do are mostly stuff that will not resolve to runtime right here (inquiry, transformational and type conversions...).
I will change the comment to reflect that for now it works with intrinsics with no optional arg.

Reply:
Improved comment in https://github.com/schweitzpgi/f18/pull/3